dbmigrate/src/main/java/se/fortnox/reactivewizard/dbmigrate/LiquibaseAutoBindModule.java

```diff
@@ -1,33 +1,31 @@
 package se.fortnox.reactivewizard.dbmigrate;
 
-import javax.inject.Inject;
-import javax.inject.Named;
-
+import com.google.inject.Binder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
 import se.fortnox.reactivewizard.binding.AutoBindModule;
-import se.fortnox.reactivewizard.config.ConfigFactory;
-import com.google.inject.Binder;
+
+import javax.inject.Inject;
+import javax.inject.Named;
 
 /**
  * Runs liquibase before the system is started, if parameter "db-migrate" is used. The following alternatives exists:
  * <ul>
- *  <li> "db-migrate": Run migrations and shutdown system. Use this locally to avoid liquibase migrations during normal run.</li>
- *  <li> "db-migrate-run": Run migrations and start the system. Use this in production, when you want to run migrations before the system is started.</li>
- *  <li> "db-drop-migrate": Drop database, run migrations and shutdown system. Use this locally to get a clean start.</li>
- *  <li> "db-drop-migrate-run": Drop database, run migrations and start the system. Use this locally to get a clean start and a running system.</li>
+ * <li> "db-migrate": Run migrations and shutdown system. Use this locally to avoid liquibase migrations during normal run.</li>
+ * <li> "db-migrate-run": Run migrations and start the system. Use this in production, when you want to run migrations before the system is started.</li>
+ * <li> "db-drop-migrate": Drop database, run migrations and shutdown system. Use this locally to get a clean start.</li>
+ * <li> "db-drop-migrate-run": Drop database, run migrations and start the system. Use this locally to get a clean start and a running system.</li>
  * </ul>
  */
 public class LiquibaseAutoBindModule implements AutoBindModule {
+    private static final Logger LOG = LoggerFactory.getLogger(LiquibaseAutoBindModule.class);
 
-    private static Logger LOG = LoggerFactory.getLogger(LiquibaseAutoBindModule.class);
-    private final LiquibaseConfig liquibaseConfig;
-    private final String startCommand;
+    private final String           startCommand;
+    private final LiquibaseMigrate liquibaseMigrate;
 
     @Inject
-    public LiquibaseAutoBindModule(ConfigFactory confFactory, @Named("args") String[] args) {
-        liquibaseConfig = confFactory.get(LiquibaseConfig.class);
+    public LiquibaseAutoBindModule(@Named("args") String[] args, LiquibaseMigrate liquibaseMigrate) {
```

Comment:
When AutoBindMoudles are run, we are in the first of two Guice iterations. At that time, https://github.com/FortnoxAB/reactive-wizard/blob/master/config/src/main/java/se/fortnox/reactivewizard/config/ConfigAutoBindModule.java has not yet bound all Config-annotated classes to it's implementation, which means that you will always get an empty implementation here. Tricky.

Reply:
Oh, I missed that. Any idea on how one could make the LiquibaseMigrate class mockable?

Reply:
You can inject a LiquibaseMigrateProvider here and let that class get a ConfigFactory to it's constructor. The ConfigFactory is bound at this time.

Reply:
That seems to have worked fine 👍